core/math/vector2.h

```diff
@@ -60,9 +60,17 @@ struct _NO_DISCARD_CLASS_ Vector2 {
 	};
 
 	_FORCE_INLINE_ real_t &operator[](int p_idx) {
+#ifdef DEBUG_ENABLED
+		ERR_FAIL_COND_V(p_idx < 0, x);
+		ERR_FAIL_COND_V(p_idx > 1, y);
```

Comment:
Another option, as I'm not sure it makes much sense to clamp like this:
```suggestion
		ERR_FAIL_INDEX_V(p_idx, 2, 0);
```
So it just returns `0` (could be `NaN`, but who likes NaNs? :P) if the index is out of size.

This should likely also be done in Vector3.

Reply:
Yes I looked at that but it's not the same as the old behaviour before the PR, so I'm suspecting in the interest of having it stable for 3.5 etc it might be better to keep the old behaviour at least to start with.

Essentially always returning 0 might introduce more bugs.

Reply:
We could incidentally modify the `ERR_FAIL_INDEX` macros to do this, as another option.

Reply:
That doesn't restore the previous behavior though, which was `p_idx ? y : x;`.

I.e. if it's 0, it's `x`, otherwise it's `y`.

Confirmed with:
```cpp
int main() {
        if (-2) {
                return 5;
        }
        return 0;
}
```
Returns `5`, if (<negative number>)` also returns `true`.

Reply:
Oh yep I see you are right. Ok I'll adjust to try and match. :+1:

Reply:
And if we add these checks to `Vector3`, there's no pre-existing bad behavior to reproduce so IMO it's better to standardize on a single "garbage" value. Using this out of bounds is a bug and will spam errors anyway, we don't need to preserve compatibility for the fallback IMO.

Reply:
Fixed. :+1: